src/devices/MatrixKeyboard/MatrixKeyboard.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Timers;
+
+namespace Iot.Device.MatrixKeyboard
+{
+    /// <summary>
+    /// GPIO Matrix Keyboard Driver
+    /// </summary>
+    public class MatrixKeyboard : IDisposable
+    {
+        /// <summary>
+        /// Get row pins
+        /// </summary>
+        public IEnumerable<int> RowPins => rowPins;
+
+        /// <summary>
+        /// Get column pins
+        /// </summary>
+        public IEnumerable<int> ColPins => colPins;
+
+        /// <summary>
+        /// Get or set scanning frequency in Hertz
+        /// </summary>
+        public double ScanFreq
+        {
+            get => 1000 / (scanTimer.Interval / colPins.Length);
+            set => scanTimer.Interval = 1000 / (value * rowPins.Length);
+        }
+
+        /// <summary>
+        /// Initialize keyboard
+        /// </summary>
+        /// <param name="gpioController">GPIO controller</param>
+        /// <param name="rowPins">Row pins</param>
+        /// <param name="colPins">Column pins</param>
+        /// <param name="scanFreq">Scanning frequency</param>
+        public MatrixKeyboard(IGpioController gpioController, IEnumerable<int> rowPins, IEnumerable<int> colPins, double scanFreq = 50)
+        {
+            gpio = gpioController;
+
+            this.rowPins = rowPins.ToArray();
+            this.colPins = colPins.ToArray();
+            buttonValues = new PinValue[this.rowPins.Length * this.colPins.Length];
+
+            scanTimer = new Timer(1000 / (scanFreq * this.rowPins.Length));
+            scanTimer.Elapsed += ScanTimerElapsed;
+        }
+
+        /// <summary>
+        /// Open GPIO pins then start keyboard scanning
+        /// </summary>
+        public void StartScan()
+        {
+            for (var i = 0; i < rowPins.Length; i++)
+            {
+                gpio.OpenPin(rowPins[i], PinMode.Output);
+            }
+            for (var i = 0; i < colPins.Length; i++)
+            {
+                gpio.OpenPin(colPins[i], PinMode.Input);
+            }
+            scanTimer.Start();
+        }
+
+        /// <summary>
+        /// Stop keyboard scanning then close GPIO pins
+        /// </summary>
+        public void StopScan()
+        {
+            scanTimer.Stop();
```

Comment:
Same here, you should check if we were actively scanning, or else it won't make much sense to do any of this work.